pyam/plotting.py

```diff
@@ -496,6 +497,67 @@ def bar_plot(df, x='year', y='value', bars='variable',
     return ax
 
 
+def boxplot(df, y='value', x='year', hue=None, palette='Spectral_r',
+            ax=None, legend=False, **kwargs):
+    """ Plot boxplot of data using seaborn.boxplot
+
+    Parameters
+    ----------
+    df : pandas.DataFrame or pyam.IamDataFrame
```

Comment:
to match other functions, this should only take a `pd.DataFrame` 

the thought here is that these utilities are purely for plotting, and internal manipulations to the `pyam.DataFrame` should happen in that scope (`core.py` or other) and not here

Reply:
agree with @gidden - let's not overload the functionality within the plotting module, this should only receive correctly formatted data from `IamDataFrame.boxplot()` or a similarly formatted pandas.DataFrame